src/transformers/models/rag/modeling_tf_rag.py

```diff
@@ -532,7 +532,32 @@ def __init__(
 
         self.question_encoder = question_encoder
         self.generator = generator
-
+    
+    # UGLY HACK: (TEMPorarily put the hack back again to show case of the new nq_checkpoint test) 
+    # really ugly fixed and require torch model: manually and ineffcient fixed of two weights name mismatched
+    @classmethod
+    def from_pretrained(cls, path_or_weight_name, model_pt=None, **kwargs):
+       
+        print(path_or_weight_name, kwargs)
+        model = super(TFRagTokenForGeneration, cls).from_pretrained(path_or_weight_name, **kwargs)
+
+        if path_or_weight_name == "facebook/rag-token-nq":
+            import gc
+            import tensorflow.keras.backend as K
+            from transformers import RagTokenForGeneration
+
+            gc.collect()
+            if model_pt is None:
+                model_pt = RagTokenForGeneration.from_pretrained(path_or_weight_name, **kwargs)
+          
+            K.set_value(model.rag.generator.model.shared.weight, model_pt.rag.generator.model.shared.weight.detach().numpy())
+            K.set_value(model.rag.generator.final_logits_bias, model_pt.rag.generator.final_logits_bias.detach().numpy())
+            del model_pt
+            gc.collect()
+            print('*** Ugly fix of weights loading -- not a generalizable solution ***')
+
+        return model
```

Comment:
For me this must be removed for two reasons:

- It needs Pytorch, the TF models have to stay outside of the PT requirements.
- using `gc` + setting manually the weights this way are a dangerous process.

To properly fix this you have to create the weights yourself in the model. Please see how it is done in the other models for that, BART and T5 are a good examples.

Reply:
@jplu Hey Julien, absolutely agrees and will remove when finalized. As commented in the thread, this is just my **temporary** communication / show-case to @patrickvonplaten to show that there's still some bug to what we together have discussed since beginning of the thread. 

> To properly fix this you have to create the weights yourself in the model. Please see how it is done in the other models for that, BART and T5 are a good examples.

And Ok! great to know this, I will try explore this direction.